packages/react-native-lockdown/src/index.js

```diff
@@ -0,0 +1,200 @@
+      )
+    }
+
+    if (new Set(callLog).size !== callLog.length) {
+      warn(
+        `LavaMoat: getRunModuleStatement was called with the same moduleId multiple times. It might be because some build is running twice with the same config instance. [${callLog.join(', ')}]`
+      )
+    }
+  }
+}
+
+/**
+ * @typedef {Object} LockdownSerializerOptions
+ * @property {boolean} [hermesRuntime=true] - Whether to enable Hermes runtime
+ *   support (otherwise JavaScript Core). Default is `true`
+ */
+
+/**
+ * Creates a Metro serializer configuration with Hardened JavaScript.
+ *
+ * @param {LockdownSerializerOptions} [options={}] - Configuration options.
+ *   Default is `{}`
+ * @param {SerializerConfig} [userConfig={}] - User-provided serializer
+ *   configuration. Default is `{}`
+ * @returns {SerializerConfig} Configured serializer options with LavaMoat
+ *   lockdown
+ */
+const lockdownSerializer = ({ hermesRuntime = true } = {}, userConfig = {}) => {
+  if (userConfig.getRunModuleStatement) {
+    warn(
+      'LavaMoat: You are using getRunModuleStatement in your serializer config. Lavamoat will attempt to wrap it without breaking it, but if you are doing something unusual, it may be affected.'
+    )
+  }
+
+  /** @type {SerializerConfig} */
+  const config = assign({}, userConfig)
+
+  /** @type {(number | string)[]} */
+  const callLog = []
+
+  /** @type {GetRunModuleStatement} */
+  const previousGetRunModuleStatement =
+    userConfig.getRunModuleStatement ?? defaultGetRunModuleStatement
+
+  const inspectCallLog = warnAboutAnomalies()
+
+  /** @type {GetRunModuleStatement} */
+  config.getRunModuleStatement = (moduleId) => {
+    callLog.push(moduleId)
+    inspectCallLog(callLog)
+    const runModuleStatement = previousGetRunModuleStatement(moduleId)
+
+    if (moduleId === ENTRY_FILE_MODULE_ID) {
+      return `hardenIntrinsics();${runModuleStatement}`
+    }
+
+    return runModuleStatement
+  }
+
+  if (!config.getPolyfills) {
+    config.getPolyfills = require('@react-native/js-polyfills')
+  } else if (typeof config.getPolyfills !== 'function') {
+    throw Error('Invalid options: getPolyfills must be a function')
+  }
+
+  const originalGetPolyfills = config.getPolyfills
+
+  /** @type {GetPolyfills} */
+  config.getPolyfills = (options) => {
+    let polyfills = originalGetPolyfills(options)
+    validatePolyfills(polyfills)
+
+    const ses = hermesRuntime
+      ? require.resolve('ses/hermes')
+      : require.resolve('ses')
+    const repair = require.resolve('./repair.js')
+
+    return [ses, repair, ...polyfills]
+  }
+
+  return config
+}
+
+/**
+ * Validates that required polyfills are correct.
+ *
+ * @param {ReadonlyArray<string>} polyfills - Array of polyfill paths
+ * @returns {void}
+ * @throws {TypeError} If required polyfills are invalid
+ */
+const validatePolyfills = (polyfills) => {
```

Comment:
```suggestion
/**
 * Validates that required polyfills are correct.
 *
 * @param {unknown} polyfills - Array of polyfill paths
 * @returns {asserts polyfills is ReadonlyArray<string>}
 * @throws {TypeError} If required polyfills are invalid
 */
const validatePolyfills = (polyfills) => {
```

Since we're checking if the value is an array, this function _technically_ accepts an unknown value.  If the function does not throw, the return type will cause the parameter to be inferred as `ReadonlyArray<string>` thereafter:


```js
const polyfills = [] // type `never[]`
validatePolyfills(polyfills)
polyfills // type `ReadonlyArray<string>`
```

> The above example begs the question, "is an empty array valid?" which you may need to answer 😄 

As a matter of convention, I would rename the function `assertPolyfills` (but that's just me).

Reply:
Oh, and the variable can be named `allegedPolyfills` 😆

Reply:
makes sense RE the types ^ updated

Reply:
perhaps `assertPolyfills` should _really_ be `assertPolyfillsAreValidForMetro`

but i think `assertPolyfills` is fine with new updated fn JSDoc description
> Asserts the alleged polyfills are valid for serialization by Metro.

Reply:
it's invalid ^ if the user provides getPolyfills that returns an empty array (Metro behaviour)

but we guide the user in _ReadMe > Existing config_ (along with an example)
> If you already used a serializer configuration, you can pass it as second argument.
`lockdownSerializer` will provide the `@react-native/js-polyfills` by default, but if you specified your own `getPolyfills` function, Metro is expecting you to provide the polyfills and `lockdownSerializer` follows that behavior too.

(and in case they don't include `@react-native/js-polyfills` like Metro expects, i added _Troubleshooting > Cannot read property 'setGlobalHandler' of undefined_)

but if worth it we can extend `assertPolyfills` to check the array that `@react-native/js-polyfills` is included